### PR TITLE
Add external catalog id and integration key to events.

### PR DESCRIPTION
**Pull request by zchenk**
- Added `external_catalog_id` and `integration_key` key/value to the following client-side events
  * Searched Site
  * Viewed Category
  * Viewed Product
  * Started Checkout
  * Added to Cart
  * Order Confirmation

## Manual Testing Steps
1. Unit tests updated for checking the event payload
2. Updated Playwright tests to validate the event properties.

## Pre-Submission Checklist:

- [ ] You've updated the CHANGELOG following the steps [here](https://github.com/klaviyo/SFCC_Klaviyo#making-updates)

**NOTE:** Please use the [Changelogger](https://pypi.org/project/changelogged/) cli tool to manage versioned file upgrades.

### CODE REVIEW

**Review by kamidzi-klaviyo-integrations — APPROVED** [review]

File: test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js [hunk 1]
```diff
@@ -1,71 +1,137 @@
...
+        testData.password = await checkoutPage.generatePassword();
+        await checkoutPage.productPage.getProduct();
+        await checkoutPage.productPage.addToCart();
+        await checkoutPage.startCheckout();
+        await checkoutPage.enterGuestEmail(email);
+        const logData = await checkoutPage.getDebugLogs(resultMsg);
+        expect(logData[0].success).toBe(true);
+
+        const events = await checkEventWithRetry(email, KLAVIYO_METRIC_ID_STARTED_CHECKOUT);
+        // console.log("Events:", JSON.stringify(events, null, 2));
```
Comment (cykolln) [inline test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:50]: nit: comment removal?

File: test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js [hunk 2]
```diff
@@ -1,71 +1,137 @@
...
+        expect(events).toBeDefined();
+        expect(events.data.length).toBeGreaterThan(0);
+
+        const metricData = events.included[0];
+        const eventData = events.data[0].attributes.event_properties;
+        const eventTime = new Date(events.data[0].attributes.datetime);
+        expect(Math.abs(new Date() - eventTime)).toBeLessThan(60 * 1000); // Within 1 minute (60,000 ms)
```
Comment (cykolln) [inline test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:57]: this mayyy flake - sometimes our APIs may not process the event in 1 min. edge case but ive seen it happen

File: test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js [hunk 3]
```diff
@@ -1,71 +1,137 @@
...
+
+        expect(metricData.attributes.name).toBe('Added to Cart');
```
Comment (cykolln) [inline test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:92]: This test may fail w/ these changes because of how the cartridge on the test site is configured. We allow either "Add to Cart" or "Added to Cart" events via the cartridge, there's a toggle on the site preferences for this. This will pass if "Send Added To Cart Event as "Add To Cart"" is set to "No" but would fail otherwise.

File: test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js [hunk 4]
```diff
@@ -1,71 +1,137 @@
...
+        await checkoutPage.productPage.addToCart();
+        await checkoutPage.startCheckout();
+        await checkoutPage.enterGuestEmail(email);
+        await checkoutPage.fillShippingForm(testData);
+        await checkoutPage.fillPaymentForm(paymentData);
+        await page.waitForTimeout(3000);
+        expect(await page.innerText('h1.page-title')).toBe('Thank You');
+
+        const events = await checkEventWithRetry(email, KLAVIYO_METRIC_ID_ORDER_CONFIRMATION);
+        // console.log("Events:", JSON.stringify(events, null, 2));
```
Comment (cykolln) [inline test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:119]: nit: comment

File: test/src/utils/klaviyo-api.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
+const axios = require('axios');
+const dotenv = require('dotenv');
+const path = require('path');
+
+// Load .env from the test directory
+const envPath = path.join(__dirname, '..', '..', '.env');
+// console.log('Loading .env from:', envPath);
```
Comment (cykolln) [inline test/src/utils/klaviyo-api.js:7]: nit comments in this file

File: test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js (hunk 1 above)
Comment (zchenk) [reply test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:50]: I intentionally left it here in case future debugging needs it.

File: test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js (hunk 3 above)
Comment (cykolln) [reply test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:92]: Not necessarily saying we need to do any refactor here, but maybe emit a log if this check fails to check the site settings. or make the test more lenient by checking for either event title.

File: test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js (hunk 2 above)
Comment (zchenk) [reply test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:57]: Isn't `event...attributes.datetime` for when the event occurred (vs when we received the message)?
For event delivery or processing delay, we could adjust the retry/wait time in `checkEventWithRetry()`.

Comment (cykolln) [reply test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:57]: Ahhh nice ok i misread - thats ok then

Comment (zchenk) [reply test/src/tests/e2e/sfra/checkout/checkout-tests.spec.js:57]: Good comments still - I updated the max wait time to 2 minutes in `checkEventWithRetry`.

**Review by cykolln — APPROVED** [review]
